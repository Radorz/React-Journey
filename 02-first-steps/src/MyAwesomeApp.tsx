import { CSSProperties } from "react";

export const MyAwesomeApp = () => {
  const firstName = "Luis";
  const lastName = "Franco";

  const favoriteGames = ["ABI", "COD", "lol"];
  const isActive = false;

  const address = {
    zipCode: "String",
    Country: "String",
  };

  const myStyles: CSSProperties = {
    backgroundColor: "red",
    borderRadius: 20,
    padding: 10,
  };

  return (
    <div data-testid="div-app">
      <h1 data-testid="first-name-title">{firstName}</h1>
      <h3>{lastName}</h3>

      <p className="mi-clase-favorita">{favoriteGames.join(", ")}</p>
      <h1>{isActive ? "Duro" : "Gay"}</h1>

      <p
        style={{
          backgroundColor: "red",
          borderRadius: 20,
          padding: 10,
        }}
      >
        {JSON.stringify(address)}
      </p>
    </div>
  );
};
