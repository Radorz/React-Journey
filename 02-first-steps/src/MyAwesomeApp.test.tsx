import { describe, expect, test } from "vitest";
import { MyAwesomeApp } from "./MyAwesomeApp";
import { render, screen } from "@testing-library/react";

describe("MyAwesomeApp", () => {
  test("Should render firstname and lastname - render", () => {
    const { container } = render(<MyAwesomeApp />);
    // screen.debug();

    const h1 = container.querySelector("h1");
    const h3 = container.querySelector("h3");

    expect(h1?.innerHTML).toContain("Luis");
    expect(h3?.innerHTML).toContain("Franco");
  });

  test("Should render firstname and lastname - screen", () => {
    render(<MyAwesomeApp />);
    screen.debug();

    const h1 = screen.getByTestId("first-name-title");
    expect(h1?.innerHTML).toContain("Luis");
  });

  test("Should match snapshot", () => {
    const { container } = render(<MyAwesomeApp />);
    expect(container).toMatchSnapshot();
  });

    test("Should match snapshot", () => {
    render(<MyAwesomeApp />);
    
    expect(screen.getByTestId('div-app')).toMatchSnapshot();
    });
});
