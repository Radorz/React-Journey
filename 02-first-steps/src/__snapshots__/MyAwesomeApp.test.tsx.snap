// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > Should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Luis
    </h1>
    <h3>
      Franco
    </h3>
    <p
      class="mi-clase-favorita"
    >
      ABI, COD, lol
    </p>
    <h1>
      Gay
    </h1>
    <p
      style="background-color: red; border-radius: 20px; padding: 10px;"
    >
      {"zipCode":"String","Country":"String"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > Should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Luis
  </h1>
  <h3>
    Franco
  </h3>
  <p
    class="mi-clase-favorita"
  >
    ABI, COD, lol
  </p>
  <h1>
    Gay
  </h1>
  <p
    style="background-color: red; border-radius: 20px; padding: 10px;"
  >
    {"zipCode":"String","Country":"String"}
  </p>
</div>
`;
